connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/ExecutionHolder.scala

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connect.service
+
+import scala.util.control.NonFatal
+
+import com.google.protobuf.Message
+import io.grpc.stub.StreamObserver
+import org.apache.commons.lang3.StringUtils
+
+import org.apache.spark.SparkSQLException
+import org.apache.spark.connect.proto
+import org.apache.spark.connect.proto.{ExecutePlanRequest, ExecutePlanResponse}
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.connect.artifact.SparkConnectArtifactManager
+import org.apache.spark.sql.connect.common.ProtoUtils
+import org.apache.spark.sql.connect.execution.{ExecutePlanResponseObserver, SparkConnectPlanExecution}
+import org.apache.spark.sql.connect.planner.SparkConnectPlanner
+import org.apache.spark.util.Utils
+
+/**
+ * Object used to hold the Spark Connect execution state, and perform
+ */
+case class ExecutionHolder(operationId: String, sessionHolder: SessionHolder) extends Logging {
+
+  val jobGroupId =
+    s"User_${sessionHolder.userId}_Session_${sessionHolder.sessionId}_Request_${operationId}"
+
+  val session = sessionHolder.session
+
+  var executePlanRequest: Option[proto.ExecutePlanRequest] = None
+
+  var executePlanResponseObserver: Option[ExecutePlanResponseObserver] = None
```

Comment:
I renamed it, and made executePlanRequest / executePlanResponseObserver options here, because I think this ExecutionHolder could be useful for holding other executions than ExecutePlanRequest.
E.g. we might want to execute requests from AnalyzePlanRequest using this as well...